server/pendinggame.js

```diff
@@ -69,33 +71,71 @@ class PendingGame {
         this.gameChat.addMessage(...arguments);
     }
 
-    join(id, user) {
-        if(_.size(this.players) === 2) {
-            return false;
-        }
-
+    addPlayer(id, user) {
         this.players[user.username] = {
             id: id,
             name: user.username,
             user: user,
             emailHash: user.emailHash,
             owner: this.owner === user.username
         };
+    }
 
-        return true;
+    newGame(id, user, password, callback) {
+        if(password) {
+            bcrypt.hash(password, 10, (err, hash) => {
+                if(err) {
+                    logger.info(err);
+
+                    callback(err);
+
+                    return;
+                }
+
+                this.password = hash;
+                this.addPlayer(id, user);
+
+                callback();
+            });
+        } else {
+            this.addPlayer(id, user);
+
+            callback();
+        }
+    }
+
+    join(id, user, password, callback) {
+        if(_.size(this.players) === 2) {
+            callback(new Error('Too many players'), 'Too many players');
+        }
+
+        if(this.password) {
+            bcrypt.compare(password, this.password, (err, valid) => {
+                if(err) {
+                    return callback(new Error('Bad password'), 'Incorrect game password');
+                }
+
+                if(!valid) {
+                    return callback(new Error('Bad password'), 'Incorrect game password');
+                }
+
+                this.addPlayer(id, user);
+
+                callback(null);
+            });
+        } else {
+            this.addPlayer(id, user);
+
+            callback(null);
+        }
     }
 
     watch(id, user) {
         if(!this.allowSpectators) {
             return false;
         }
 
-        this.spectators[user.username] = {
-            id: id,
-            name: user.username,
-            emailHash: user.emailHash,
-            user: user
-        };
+        this.addPlayer(id, user);
```

Comment:
If I'm reading this correct, this will count spectators toward the player count. So if I start a game and then a spectator watches, no one else can join as a player and I could start the game with that spectator. Even weirder, if another spectator watches, a player COULD join since the failure check is `_.size(this.players) === 2`, at which point you could start the game with 3+ players.

Additionally, you updated the usage of `PendingGame.watch` in the lobby to take a password and callback like `join`, but haven't modified here to implement that.